ecommerce/extensions/payment/views/lumsxpay.py

```diff
@@ -96,10 +89,11 @@ def request_already_existing_challan(self, request):
         return {}
 
     def get(self, request):
+        basket = request.basket
         configuration_helpers = request.site.siteconfiguration.edly_client_theme_branding_settings
         url = configuration_helpers.get('LUMSXPAY_VOUCHER_API_URL')
 
-        if request.user.is_anonymous or not (url and request.basket):
+        if request.user.is_anonymous or not (url and basket):
```

Comment:
why not wrap inside `login_required` decorator?

Reply:
wont it be overhead as we only need to check anon person and other custom conditions

Reply:
can be tried

Reply:
For classes, we can use `LoginRequiredMixin`